src/libraries/Microsoft.PowerFx.Core/Parser/ParseNamedFormulasAndUDFResult.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Errors;
+
+namespace Microsoft.PowerFx.Core.Parser
+{
+    internal class ParseNamedFormulasAndUDFResult
+    {
+        internal IEnumerable<UDF> UDFs { get; }
+
+        internal IEnumerable<NamedFormula> NamedFormulas { get; }
+
+        internal IEnumerable<TexlError> Errors { get; }
+
+        internal bool HasErrors { get; }
+
+        public ParseNamedFormulasAndUDFResult(IEnumerable<NamedFormula> namedFormulas, IEnumerable<UDF> uDFs, IEnumerable<TexlError> errors)
```

Comment:
Now that we have this concept, do we need the old ParseUDFs result in isolation?

Reply:
We don't for now, but in future if we need a way to parse UDFs separately, we might need that.